src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -127,17 +127,45 @@ public function getMaxDepth(): ?int
     /**
      * {@inheritdoc}
      */
-    public function setSerializedName(string $serializedName = null)
+    public function setSerializedNames(array $serializedNames): void
     {
-        $this->serializedName = $serializedName;
+        $this->serializedNames = $serializedNames;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getSerializedName(): ?string
+    public function setSerializedName(string $serializedName = null, array $groups = [])
     {
-        return $this->serializedName;
+        if (empty($groups)) {
+            $this->serializedNames['*'] = $serializedName;
+        }
+
+        foreach ($groups as $group) {
+            $this->serializedNames[$group] = $serializedName;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSerializedNames(): array
+    {
+        return $this->serializedNames;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSerializedName(array $groups = []): ?string
```

Comment:
Do we really need to provide an array here?
I mean, is there any use case where you need to get a serialized name for more than one group?

If not, you can write instead `getSerializedName(string $group = null): ?string`

WDYT?

Reply:
`$data = $serializer->normalize($obj, null, ['groups' => ['groupA', 'groupB]);`
`MetadataAwareNameConverter` will use theses groups to determine which serializedName to use.

I think array is required

Reply:
Indeed! Thanks :slightly_smiling_face: